patches/expo-secure-store/expo-secure-store+14.2.4+001+additional-config-options.patch

```diff
@@ -0,0 +1,1114 @@
++     * @platform android
++     */
++    BIOMETRICS: 2,
++    /**
++     * FaceID was used to authenticate
++     * @platform ios
++     */
++    FACE_ID: 3,
++    /**
++     * TouchID was used to authenticate
++     * @platform ios
++     */
++    TOUCH_ID: 4,
++    /**
++     * OpticID was used to authenticate (reserved by apple, used on Apple Vision Pro, not iOS)
++     */
++    OPTIC_ID: 5
++};
+ /**
+  * The data in the keychain item cannot be accessed after a restart until the device has been
+  * unlocked once by the user. This may be useful if you need to access the item when the phone
+@@ -102,7 +148,7 @@ export async function setItemAsync(key, value, options = {}) {
+     if (!isValidValue(value)) {
+         throw new Error(`Invalid value provided to SecureStore. Values must be strings; consider JSON-encoding your values if they are serializable.`);
+     }
+-    await ExpoSecureStore.setValueWithKeyAsync(value, key, options);
++    return await ExpoSecureStore.setValueWithKeyAsync(value, key, options);
+ }
+ /**
+  * Stores a key–value pair synchronously.
+diff --git a/node_modules/expo-secure-store/ios/SecureStoreModule.swift b/node_modules/expo-secure-store/ios/SecureStoreModule.swift
+index 439b08d..c02be32 100644
+--- a/node_modules/expo-secure-store/ios/SecureStoreModule.swift
++++ b/node_modules/expo-secure-store/ios/SecureStoreModule.swift
+@@ -4,6 +4,37 @@ import LocalAuthentication
+ #endif
+ import Security
+ 
++@available(iOS 11.0, *)
++public enum AuthType: Int, @unchecked Sendable {
++  /// The device does not support biometry.
++  @available(iOS 11.2, *)
++  case none = 0
++
++  case credentials = 1
++
++  /// The device supports Face ID.
++  case faceID = 3
++
++  /// The device supports Touch ID.
++  case touchID = 4
++
++  /// The device supports Optic ID
++  @available(iOS 17.0, *)
++  case opticID = 5
++}
++
++struct RuntimeError: LocalizedError {
++  let description: String
++
++  init(_ description: String) {
++    self.description = description
++  }
++
++  var errorDescription: String? {
++    description
++  }
++}
++
+ public final class SecureStoreModule: Module {
+   public func definition() -> ModuleDefinition {
+     Name("ExpoSecureStore")
+@@ -18,57 +49,163 @@ public final class SecureStoreModule: Module {
+       "WHEN_UNLOCKED_THIS_DEVICE_ONLY": SecureStoreAccessible.whenUnlockedThisDeviceOnly.rawValue
+     ])
+ 
+-    AsyncFunction("getValueWithKeyAsync") { (key: String, options: SecureStoreOptions) -> String? in
+-      return try get(with: key, options: options)
++    AsyncFunction("getValueWithKeyAsync") { (key: String, options: SecureStoreOptions) -> [Any] in
++      let returnValue = try get(with: key, options: options)
++
++      if options.requireAuthentication && options.authOnEveryAction {
++        try await askForAuth(
++          prompt: options.authenticationPrompt
++          ?? "Authentication required"
++        )
++      }
```

Comment:
I suggest moving this guard over ` let returnValue = try get(with: key, options: options)` for safety reasons